new-pipeline/src/dashboards.js

```diff
@@ -163,8 +163,14 @@ $(document)
           async:false
         });
         document.execCommand('copy');
+        });
       });
-   });
+
+// Work around to force the clipboard to hold short URL from permalink button.
+document.addEventListener("copy", e => {
+  e.clipboardData.setData("text/plain", shortUrl);
```

Comment:
`shortUrl` won't be present in this context, since it only lives about as long as the `success` handler from line 152 is running. 

Since the success handler sets the value of the `input` to `shortUrl`, you should be able to read it from there at the point in time the copy event listener runs.

Make sense?

Reply:
Yes, thank you!

Reply:
By the way, do the extra indents seem to be corrected now?

Reply:
The indents seem better. There are two little indents that need fixing, though.

Lines 136-139 should be two spaces further in (`.mouseup` up to the closing `)};`)
And Lines 166 and 167 should be two spaces further back (the two `)};` should be indented on the original level, not inline with the method bodies.

So it should look like
```js
this.doSomething()
  .andSomethingElse(function () {
    theFunctionBodyGoesHere();
  }); //should line up with "." instead of with "the"
```